packages/iris-grid/src/IrisGrid.jsx

```diff
@@ -1371,6 +1374,34 @@ export class IrisGrid extends Component {
     this.grid.forceUpdate();
   }
 
+  freezeColumnByColumnName(columnName) {
+    const { frozenColumns } = this.state;
+    const { model } = this.props;
+    log.debug2('freezing column', columnName);
+
+    let allFrozenColumns;
+
+    if (frozenColumns == null) {
+      allFrozenColumns = new Set(model.layoutHints?.frozenColumns ?? []);
+      allFrozenColumns.add(columnName);
+    } else {
+      allFrozenColumns = new Set([...frozenColumns, columnName]);
+    }
+
+    this.setState({
+      frozenColumns: [...allFrozenColumns],
+    });
+  }
+
+  unFreezeColumnByColumnName(columnName) {
+    const { frozenColumns } = this.state;
+    log.debug2('unfreezing column', columnName);
+
```

Comment:
You need the `allFrozenColumns` thing like you do in `freezeColumnByColumnName`. Otherwise if the first operation is an unfreeze on a table with frozen columns in layout hints, boom.